### PR TITLE
vendor: update kvproto

### PR DESCRIPTION
**Pull request by tiancaiamao**
go-grpc-middleware need to be locked, because it specify a different version of grpc.
I'd like to update kvproto for https://github.com/pingcap/tidb/pull/6457, and `make update` get me here.

PTAL @zhexuany  @coocood

### CODE REVIEW

**Comment by tiancaiamao** [comment]
So I just update kvproto now, manually call

```
dep ensure -update github.com/pingcap/kvproto
dep prune
./hack/clean_vendor.sh
```

It seems somebody don't make the right update last time, so some extra files is included after this operation.

**Comment by shenli** [comment]
But the kvproto in vendor does not changed.

**Comment by tiancaiamao** [comment]
@shenli 
Of course it's changed:
https://github.com/pingcap/tidb/pull/6461/files#diff-f245df97eef01edcf335da839deca155R217

File: vendor/github.com/pingcap/kvproto/pkg/kvrpcpb/kvrpcpb.pb.go [hunk 1]
```diff
@@ -211,9 +211,10 @@ func (m *LockInfo) GetLockTtl() uint64 {
 }
 
 type KeyError struct {
-	Locked    *LockInfo `protobuf:"bytes,1,opt,name=locked" json:"locked,omitempty"`
-	Retryable string    `protobuf:"bytes,2,opt,name=retryable,proto3" json:"retryable,omitempty"`
-	Abort     string    `protobuf:"bytes,3,opt,name=abort,proto3" json:"abort,omitempty"`
+	Locked    *LockInfo      `protobuf:"bytes,1,opt,name=locked" json:"locked,omitempty"`
+	Retryable string         `protobuf:"bytes,2,opt,name=retryable,proto3" json:"retryable,omitempty"`
+	Abort     string         `protobuf:"bytes,3,opt,name=abort,proto3" json:"abort,omitempty"`
+	Conflict  *WriteConflict `protobuf:"bytes,4,opt,name=conflict" json:"conflict,omitempty"`
```
Comment (tiancaiamao) [inline vendor/github.com/pingcap/kvproto/pkg/kvrpcpb/kvrpcpb.pb.go]: It's changed @shenli

**Comment by coocood** [comment]
LGTM

**Pull request merged**